File: src/main/java/com/academy/model/Payment.java
package com.academy.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "payments")
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long paymentId;

    @ManyToOne
    @JoinColumn(name = "kid_id", nullable = false)
    private Kid kid;

    @ManyToOne
    @JoinColumn(name = "invoice_id")
    private FeeInvoice feeInvoice; // Link to specific invoice

    @Column(nullable = false)
    private Double amount;

    @Column(nullable = false)
    private LocalDateTime paymentDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentMethod paymentMethod;

    @Column(unique = true)
    private String transactionId;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    public enum Status {
        PENDING, COMPLETED, FAILED
    }

    public enum PaymentMethod {
        CASH, CARD, MOBILE
    }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: src/main/resources/football-academy-frontend/src/app/models/payment.ts
import { CashbookDTO } from "./cashbook-dto";
import { FeeInvoice } from "./fee-invoice";
import { Kid } from "./kid";

export interface Payment {
  paymentId: number;
  kid: Kid;
  amount: number;
  paymentDate: string;
  bankingDate?: string;
  status: 'PENDING' | 'COMPLETED' | 'FAILED';
  paymentMethod: 'CASH' | 'CARD' | 'MOBILE' | 'BANK_TRANSFER' | 'DIGITAL_WALLET';
  transactionId?: string;
  cashbook: CashbookDTO;
  overpaymentAmount: number;
  createdAt: string;
  updatedAt: string;
}